netket/jax/_expect.py

```diff
@@ -33,33 +33,51 @@ def expect(
     σ: jnp.ndarray,
     *expected_fun_args,
     n_chains: int = None,
+    _return_estimators: bool = False,
 ) -> Tuple[jnp.ndarray, Stats]:
     """
     Computes the expectation value over a log-pdf.
-
-    Args:
-        log_pdf:
-        expected_ffun
     """
-    return _expect(n_chains, log_pdf, expected_fun, pars, σ, *expected_fun_args)
+    return _expect(
+        n_chains, log_pdf, expected_fun, pars, σ, _return_estimators, *expected_fun_args
+    )
 
 
 # log_prob_args and integrand_args are independent of params when taking the
 # gradient. They can be continuous or discrete, and they can be pytrees
 # Does not support higher-order derivatives yet
-@partial(jax.custom_vjp, nondiff_argnums=(0, 1, 2))
-def _expect(n_chains, log_pdf, expected_fun, pars, σ, *expected_fun_args):
+@partial(jax.custom_vjp, nondiff_argnums=(0, 1, 2, 5))
+def _expect(
+    n_chains,
+    log_pdf,
+    expected_fun,
+    pars,
+    σ,
+    _return_estimators: bool,
+    *expected_fun_args,
+):
     L_σ = expected_fun(pars, σ, *expected_fun_args)
     if n_chains is not None:
         L_σ = L_σ.reshape((n_chains, -1))
 
     L̄_σ = mpi_statistics(L_σ.T)
     # L̄_σ = L_σ.mean(axis=0)
 
-    return L̄_σ.mean, L̄_σ
+    if _return_estimators:
+        return L̄_σ.mean, (L̄_σ, L_σ.T)
```

Comment:
```suggestion
        return L̄_σ.mean, L̄_σ, L_σ.T
```
This output always goes through a tuple unpacking anyway, why make that more complicated?

Reply:
I think there was some issue with `vjp` when calling it without grouping the non-differentiated return values in a single tuple.